scripts/rename_docker_images.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+import argparse as _ap
+import datetime as _dt
+import os
+import pathlib as _pl
+import subprocess as _sp
+
+MERGE_REQUEST_GROOVY = "L0_MergeRequest.groovy"
+IMAGE_MAPPING = {
+    "LLM_DOCKER_IMAGE":
+    "urm.nvidia.com/sw-tensorrt-docker/tensorrt-llm-staging/devel:x86_64-devel-torch_skip",
+    "LLM_SBSA_DOCKER_IMAGE":
+    "urm.nvidia.com/sw-tensorrt-docker/tensorrt-llm-staging/devel:sbsa-devel-torch_skip",
+    "LLM_ROCKYLINUX8_PY310_DOCKER_IMAGE":
+    "urm.nvidia.com/sw-tensorrt-docker/tensorrt-llm-staging/devel:x86_64-rockylinux8-torch_skip-py310",
+    "LLM_ROCKYLINUX8_PY312_DOCKER_IMAGE":
+    "urm.nvidia.com/sw-tensorrt-docker/tensorrt-llm-staging/devel:x86_64-rockylinux8-torch_skip-py312",
+}
+
+BUILD_GROOVY = "Build.groovy"
+
+SRC_PATTERN = "urm.nvidia.com/sw-tensorrt-docker/tensorrt-llm-staging/devel:x86_64-devel-torch_skip"
+# [base_image_name]-[arch]-[os](-[python_version])-[trt_version]-[torch_install_type]-[stage]-[date]-[mr_id]
+DST_IMAGE = "LLM_DOCKER_IMAGE"
+
+
+def parse_arguments() -> _ap.Namespace:
+    parser = _ap.ArgumentParser(
+        description="Rename Docker images based on the given instructions.")
+    parser.add_argument(
+        'src_branch',
+        type=str,
+        help="The name of the source branch releasing the Docker image.")
+    parser.add_argument(
+        'src_build_id',
+        type=int,
+        help="The name of the source build id release the Docker image.")
+    parser.add_argument(
+        'dst_mr',
+        type=int,
+        help="The number of the merge request for the destination image.")
+    parser.add_argument(
+        '--dry-run',
+        action='store_true',
+        help="Simulate the rename process without making any changes.")
+    parser.add_argument(
+        "--timestamp",
+        type=str,
+        required=False,
+        help="The timestamp to use for the destination image name.")
+    return parser.parse_args()
+
+
+def get_current_timestamp() -> str:
+    """Get the current timestamp in YYYYMMDDhhmm format."""
+    return _dt.datetime.now(_dt.UTC).strftime("%Y%m%d%H%M")
```

Comment:
Thank you for providing the script. I have conducted experiments locally. Since the local `datetime` version is older, I will make modifications to this part and use `datetime.timezone.utc`.
https://github.com/python/cpython/pull/91973